plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -580,14 +583,15 @@ export class SessionRecordingIngester {
 
     private async reportPartitionMetrics() {
         /**
-         * For all partitions we are assigned, report metrics. Otherwise clear them.
+         * For all partitions we are assigned, report metrics.
+         * For any other number we clear the metrics from our gauges
          */
         const assignedPartitions = this.assignedTopicPartitions.map((x) => x.partition)
         const offsetsByPartition = await this.latestOffsetsRefresher.get()
 
-        Object.entries(this.partitionMetrics).forEach(([partitionString, metrics]) => {
-            const partition = parseInt(partitionString)
+        for (let partition = 0; partition < this.totalNumPartitions; partition++) {
```

Comment:
could `this.totalNumPartitions` be `env.TOTAL_PARTITIONS` or tbh `128`

is it expensive enough to clear partitions metrics that don't exist as compared to frequently asking kafka what partitions it has

(I can't remember which but changing partitions up or down is hard so we're likely to change partitions infrequently)

Reply:
I guess so... Just means we have to have a custom var, wherever it is deployed. This only runs on startup so is really lightweight

Reply:
feels simpler to me to hard code a very slow changing value like this... but no strong feeling on this... better to have the metrics correct and come back to this if we choose to 👍